lighthouse-core/scripts/build-report-for-autodeployment.js

```diff
@@ -49,3 +48,68 @@ lhr.categories['lighthouse-plugin-someplugin'] = {
     }
   });
 })();
+
+/**
+ * Add a plugin to demo plugin rendering.
+ * @param {LH.Result} lhr
+ */
+function addPluginCategory(lhr) {
+  lhr.categories['lighthouse-plugin-someplugin'] = {
+    id: 'lighthouse-plugin-someplugin',
+    title: 'Plugin',
+    score: 0.5,
+    auditRefs: [],
+  };
+}
+
+/**
+ * Generate an LHR with errors for the renderer to display
+ * We'll write an "empty" artifacts file to disk, only to use it in auditMode
+ * @return {Promise<LH.Result>}
+ */
+async function generateErrorLHR() {
+  const url = 'http://fakeurl.com';
+  const options = {
+    requestedUrl: url,
+    settings: defaultSettings,
+    driver: {
+      getBrowserVersion: () => ({userAgent: 'Mozilla/5.0 ErrorUserAgent Chrome/66'}),
+    },
+  };
+  // @ts-ignore driver isn't mocked out completely
+  const artifacts = await GatherRunner.initializeBaseArtifacts(options);
+  // Add in a global runWarning
+  artifacts.LighthouseRunWarnings.push(`Something went wrong with recording the trace over your
+  page load. Please run Lighthouse again. (NO_FCP)`);
+
+  // Save artifacts to disk then run `lighthouse -G` with them
+  const TMP = `${DIST}/.tmp/`;
+  mkdirp(TMP);
+  fs.writeFileSync(`${TMP}/artifacts.json`, JSON.stringify(artifacts), 'utf-8');
+  const errorRunnerResult = await lighthouse(url, {auditMode: TMP});
+  const errorLhr = /** @type {LH.RunnerResult} */ (errorRunnerResult).lhr;
```

Comment:
Why does this need a cast?

Reply:
because `lighthouse` can return `RunnerResult | undefined` (i think because of -G?)